src/components/Map/utils.ts

```diff
@@ -83,8 +83,20 @@ export const getMapDrawingLayer = (source: MapSourceRenderProperties, sourceId:
       id: `${sourceId}-outline`,
       type: 'line',
       paint: {
-        'line-color': source.lineColor,
-        'line-width': source.lineWidth,
+        'line-color': [
+          // if feature is selected, use selectLineColor if defined
+          'case',
+          ['boolean', ['feature-state', 'select'], false],
+          source.selectLineColor ?? source.lineColor,
+          source.lineColor,
```

Comment:
line 90 is active when feature-state 'select' is true on an object, line 91 is the 'else' equivalent
see https://docs.mapbox.com/style-spec/reference/expressions/#case